page/ajax/working-with-jsonp.md

```diff
@@ -5,30 +5,30 @@
 	"attribution": [ "jQuery Fundamentals" ]
 }</script>
 
-The advent of JSONP — essentially a consensual cross-site scripting hack — has opened the door to powerful mashups of content. Many prominent sites provide JSONP services, allowing you access to their content via a predefined API. A particularly great source of JSONP-formatted data is the [Yahoo! Query Language](http://developer.yahoo.com/yql/console/), which we'll use in the following example to fetch news about cats.
+The advent of JSONP — essentially a consensual cross-site scripting hack — has opened the door to powerful mashups of content. Many prominent sites provide JSONP services, allowing you access to their content via a predefined API. A particularly great source of JSONP-formatted data is the [Wikipedia API](https://en.wikipedia.org/w/api.php?action=help&modules=query), which we'll use in the following example to fetch a list of articles about [JavaScript libraries](https://en.wikipedia.org/wiki/Category:Free_software_programmed_in_JavaScript).
 
 ```
-// Using YQL and JSONP
 $.ajax({
-	url: "http://query.yahooapis.com/v1/public/yql",
+	// Tell MediaWiki what we want and that we want JSON
+	url: "https://en.wikipedia.org/w/api.php",
+	data: {
+		format: "json",
+		action: "query",
+		list: "categorymembers",
+		cmtitle: "Category:Free_software_programmed_in_JavaScript",
+		cmprop: "title",
+		cmlimit: 500
+	},
 
-	// The name of the callback parameter, as specified by the YQL service
+	// Name of the query parameter that jQuery will add, to set the callback function
 	jsonp: "callback",
 
 	// Tell jQuery we're expecting JSONP
 	dataType: "jsonp",
 
-	// Tell YQL what we want and that we want JSON
-	data: {
-		q: "select title,abstract,url from search.news where query=\"cat\"",
-		format: "json"
-	},
-
-	// Work with the response
-	success: function( response ) {
-		console.log( response ); // server response
-	}
+}).then(function( response ) {
+	console.log( response.query.categorymembers );
 });
 ```
 
-jQuery handles all the complex aspects of JSONP behind-the-scenes — all we have to do is tell jQuery the name of the JSONP callback parameter specified by YQL ("callback" in this case), and otherwise the whole process looks and feels like a normal Ajax request.
+jQuery handles all the complex aspects of JSONP behind-the-scenes all we have to do is tell jQuery the name of the JSONP callback parameter ("callback" in this case). Beyond that one option, the whole process looks and feels like a normal Ajax request for fetching JSON.
```

Comment:
The hyphen before “all we have to do” was needed here.

Also, the `jsonp` option is no longer the only required option. In jQuery 4, you’ll also need to explicitly specify `dataType: "jsonp"`. The example above already does it but let’s list it here as a requirement.

Reply:
I agree that it needs some punctuation. Instead of a hyphen (or in this case, an em-dash), I'd suggest a semicolon or a period.